AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_processor_batch_tests.py

```diff
@@ -330,41 +326,42 @@ def test_TwoAssetsWithSameProductName_ShouldProcessAfterRename(self, asset_proce
         """
 
         def run_ap_debug_skip_atom_output(asset_processor):
-            result, output = asset_processor.batch_process(capture_output=True, extra_params=["--debugOutput",
-                                                                                              "--regset=\"/O3DE/SceneAPI/AssetImporter/SkipAtomOutput=true\""])
+            result, output = asset_processor.batch_process(capture_output=True, extra_params=["--regset=\"/O3DE/SceneAPI/AssetImporter/SkipAtomOutput=true\""])
             # If the test fails, it's helpful to have the output from asset processor in the logs, to track the failure down.
             logger.info(f"Asset Processor Output: {pformat(output)}")
             return result, output
 
         # Copying test assets to project folder
         asset_processor.prepare_test_environment(ap_setup_fixture["tests_dir"], "test_TwoAssetsWithSameProductName_ShouldProcessAfterRename")
+
         # Launching AP, verify it is failing
         result, output = run_ap_debug_skip_atom_output(asset_processor)
         assert result == False, \
             'AssetProcessorBatch should have failed because the generated output products should share the same name.'
 
         # Renaming output files so they won't collide in cache after second processing
-        file_to_rename = os.path.join(asset_processor.temp_asset_root(), "AutomatedTesting", "test_TwoAssetsWithSameProductName_ShouldProcessAfterRename", "a.fbx.assetinfo")
+        asset_info_file = os.path.join(asset_processor.temp_asset_root(), "AutomatedTesting", "test_TwoAssetsWithSameProductName_ShouldProcessAfterRename", "a.fbx.assetinfo")
         data_for_rename = os.path.join(asset_processor.temp_asset_root(), "AutomatedTesting", "test_TwoAssetsWithSameProductName_ShouldProcessAfterRename", "rename.txt")
 
-        assert os.path.exists(file_to_rename), "An assetinfo file is missing."
-        assert os.path.exists(data_for_rename), "Text file to copy into the assetinfo file is missing."
+        os.remove(asset_info_file)
```

Comment:
`os.remove` will raise an exception if the file does not currently exist on disk, is this intentional? The previous code was asserting that the path existed with a helpful message before attempting to modify it.

Reply:
I was assuming the file always existed as it's one of the test assets copied in, we wouldn't actually get this far in the test if the file was missing actually as AP would pass when it should have failed -  the assetinfo file it's referring to is what causes AP to intentionally fail in the step before. 
I could still add it for clarity though, thanks for the call out!